axum-login/src/service.rs

```diff
@@ -86,6 +86,13 @@ where
                     }
                 };
 
+                // Provide the user as a request extension when it's available.
+                if let Some(ref user) = auth_session.user {
+                    tracing::Span::current().record("user.id", &user.id().to_string());
+                    req.extensions_mut().insert(user.clone());
```

Comment:
The clone here might be worth considering: if an application never uses this request extension and instead uses the AuthSession directly then this is unnecessary overhead.